### PR TITLE
Update selectors to handle optional optimism

### PR DESCRIPTION
**Pull request by Chochek**

### CODE REVIEW

**Comment by Chochek** [comment]
[(3) Update redux-clerk selectors to handle optional optimism](https://trello.com/c/29zEwd1Q/116-3-update-redux-clerk-selectors-to-handle-optional-optimism)

File: src/selectors/dataset.js [hunk 1]
```diff
@@ -14,16 +14,91 @@ export const datasetSelector = (config, state, instance) => {
...
+  let instanceData = baseState.getIn(['instances', instance, 'data'])
   const rawData = baseState.get('raw')
 
   // If instanceData is not created yet, return empty Map
   if(!instanceData) {
     return Map({})
   }
 
+  // Remove the pending.create items from the instance
+  instanceData = instanceData.filter(uid => baseState.getIn(['pending', 'create'], List()).indexOf(uid) === -1)
```
Comment (jivinivan) [inline src/selectors/dataset.js]: We should only get `baseState.getIn(['pending', 'create'], List()` once above set in a const. That way we aren't getting the pending create items inside a loop.

Something like:
```
const pendingCreate = baseState.getIn(['pending', 'create'], List()
instanceData = instanceData.filter(uid => pendingCreate.indexOf(uid) === -1)
```

File: src/selectors/dataset.js [hunk 2]
```diff
@@ -14,16 +14,91 @@ export const datasetSelector = (config, state, instance) => {
...
+      // Check pending.update and add the pendingUpdate prop if necessary
+      const pendingUpdate = baseState.getIn(['pending', 'update'], List()).indexOf(uid) > -1
+      if (pendingUpdate) {
+        map.set('pendingUpdate', true)
+      }
+      // Check pending.remove and add the pendingRemove prop if necessary
+      const pendingRemove = baseState.getIn(['pending', 'remove'], List()).indexOf(uid) > -1
+      if (pendingRemove) {
+        map.set('pendingRemove', true)
+      }
```
Comment (jivinivan) [inline src/selectors/dataset.js]: Get `baseState.getIn(['pending', 'update'], List())` and `baseState.getIn(['pending', 'remove'], List())` outside of loop.

File: src/selectors/dataset.js [hunk 3]
```diff
@@ -14,16 +14,91 @@ export const datasetSelector = (config, state, instance) => {
...
+
+  let rawData = baseState.get('raw')
+
+  // If instanceData is not created yet, return empty Map
+  if(!instanceData) {
+    return Map({})
+  }
+
+  // Remove the pending.remove items
+  instanceData = instanceData.filter(uid => baseState.getIn(['pending', 'remove'], List()).indexOf(uid) === -1)
```
Comment (jivinivan) [inline src/selectors/dataset.js]: Get `baseState.getIn(['pending', 'remove'], List())` outside of loop.

File: src/selectors/dataset.js [hunk 4]
```diff
@@ -14,16 +14,91 @@ export const datasetSelector = (config, state, instance) => {
...
+      // Check pending.create and add the pendingCreate prop if necessary
+      const pendingCreate = baseState.getIn(['pending', 'create'], List()).indexOf(uid) > -1
+      if (pendingCreate) {
+        map.set('pendingCreate', true)
+      }
+      // Check pending.update and add the pendingUpdate prop if necessary
+      const pendingUpdate = baseState.getIn(['pending', 'update'], List()).indexOf(uid) > -1
+      if (pendingUpdate) {
+        map.set('pendingUpdate', true)
+      }
```
Comment (jivinivan) [inline src/selectors/dataset.js]: Get `baseState.getIn(['pending', 'update'], List())` and `baseState.getIn(['pending', 'remove'], List())` outside of loop.

**Pull request merged**